{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Aula 11: Análise de variantes.ipynb",
      "provenance": [],
      "authorship_tag": "ABX9TyOg3HZCGeDtTrz2omx5Przq",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/mariacmartins/bioinformatica-disciplina/blob/main/Aula_11_An%C3%A1lise_de_variantes.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Jj00SEoHJY4l"
      },
      "source": [
        "# Análise de variantes\n",
        "\n",
        "No processo de identificação de variantes com NGS, por exemplo, o resultado do alinhamento é processado de modo a se identificar posições do genoma onde há **divergência** entre o genoma de interesse e o genoma de referência. \n",
        "\n",
        "**Conceitos importantes:**\n",
        "- **SNPS:** Polimorfismo de nucleotídeo único (em inglês single nucleotide polymorphism) é uma variação na sequência de DNA que afeta somente uma base (adenina, timina, citosina ou guanina) na sequência do genoma entre indivíduos de uma espécie ou entre pares de cromossomos de um individuo. \n",
        "- **INDELS:** Indel é um termo da biologia molecular utilizado para classificar pequenas variações genéticas, sendo essas inserções ou deleções de bases no genoma de um organismo. \n",
        "- **Arquivo VCF:** O Variant Call Format (VCF) especifica o formato de um arquivo de texto usado em bioinformática para armazenar variações de sequência de genes.\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "l1PWN4Tmep_B"
      },
      "source": [
        "#### Instalação dos pacotes e ferramentas"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jVYtlXtaIxjK",
        "outputId": "2a2f864d-2e37-4d3d-b532-e5c1979e2bdd"
      },
      "source": [
        "!apt install sra-toolkit bwa samtools ncbi-entrez-direct bcftools\n",
        "!pip install snptoolkit"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  blends-common cwltool libauthen-sasl-perl libc-ares2 libdata-dump-perl\n",
            "  libencode-locale-perl libfile-listing-perl libfont-afm-perl\n",
            "  libhtml-form-perl libhtml-format-perl libhtml-parser-perl\n",
            "  libhtml-tagset-perl libhtml-tree-perl libhts2 libhttp-cookies-perl\n",
            "  libhttp-daemon-perl libhttp-date-perl libhttp-message-perl\n",
            "  libhttp-negotiate-perl libio-html-perl libio-socket-ssl-perl libjs-bootstrap\n",
            "  libkdf5-2 liblwp-mediatypes-perl liblwp-protocol-https-perl libmagic-mgc\n",
            "  libmagic1 libmailtools-perl libmbedcrypto1 libmbedtls10 libncbi-vdb2\n",
            "  libncbi-wvdb2 libnet-http-perl libnet-smtp-ssl-perl libnet-ssleay-perl\n",
            "  libtimedate-perl libtry-tiny-perl liburi-perl libwww-perl\n",
            "  libwww-robotrules-perl med-config menu netbase nodejs nodejs-doc\n",
            "  perl-openssl-defaults python-asn1crypto python-avro python-cachecontrol\n",
            "  python-certifi python-cffi-backend python-chardet python-cryptography\n",
            "  python-enum34 python-html5lib python-idna python-ipaddress python-isodate\n",
            "  python-lockfile python-mistune python-openssl python-pkg-resources\n",
            "  python-pyparsing python-rdflib python-rdflib-jsonld python-requests\n",
            "  python-ruamel.yaml python-schema-salad python-shellescape python-six\n",
            "  python-sparqlwrapper python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  python-matplotlib texlive-latex-recommended blends-doc libdigest-hmac-perl\n",
            "  libgssapi-perl libcrypt-ssleay-perl file libauthen-ntlm-perl menu-l10n gksu\n",
            "  | kde-runtime | ktsuss python-cryptography-doc python-cryptography-vectors\n",
            "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
            "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
            "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  bcftools blends-common bwa cwltool libauthen-sasl-perl libc-ares2\n",
            "  libdata-dump-perl libencode-locale-perl libfile-listing-perl\n",
            "  libfont-afm-perl libhtml-form-perl libhtml-format-perl libhtml-parser-perl\n",
            "  libhtml-tagset-perl libhtml-tree-perl libhts2 libhttp-cookies-perl\n",
            "  libhttp-daemon-perl libhttp-date-perl libhttp-message-perl\n",
            "  libhttp-negotiate-perl libio-html-perl libio-socket-ssl-perl libjs-bootstrap\n",
            "  libkdf5-2 liblwp-mediatypes-perl liblwp-protocol-https-perl libmagic-mgc\n",
            "  libmagic1 libmailtools-perl libmbedcrypto1 libmbedtls10 libncbi-vdb2\n",
            "  libncbi-wvdb2 libnet-http-perl libnet-smtp-ssl-perl libnet-ssleay-perl\n",
            "  libtimedate-perl libtry-tiny-perl liburi-perl libwww-perl\n",
            "  libwww-robotrules-perl med-config menu ncbi-entrez-direct netbase nodejs\n",
            "  nodejs-doc perl-openssl-defaults python-asn1crypto python-avro\n",
            "  python-cachecontrol python-certifi python-cffi-backend python-chardet\n",
            "  python-cryptography python-enum34 python-html5lib python-idna\n",
            "  python-ipaddress python-isodate python-lockfile python-mistune\n",
            "  python-openssl python-pkg-resources python-pyparsing python-rdflib\n",
            "  python-rdflib-jsonld python-requests python-ruamel.yaml python-schema-salad\n",
            "  python-shellescape python-six python-sparqlwrapper python-typing\n",
            "  python-urllib3 python-webencodings samtools sra-toolkit\n",
            "0 upgraded, 79 newly installed, 0 to remove and 34 not upgraded.\n",
            "Need to get 15.6 MB of archives.\n",
            "After this operation, 73.4 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagic-mgc amd64 1:5.32-2ubuntu0.4 [184 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libmagic1 amd64 1:5.32-2ubuntu0.4 [68.6 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/main amd64 netbase all 5.4 [12.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bcftools amd64 1.7-2 [503 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 menu amd64 2.1.47ubuntu2.1 [349 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/universe amd64 blends-common all 0.6.100ubuntu2 [15.0 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 bwa amd64 0.7.17-1ubuntu0.1 [176 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 libdata-dump-perl all 1.23-1 [27.0 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 libencode-locale-perl all 1.05-1 [12.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/main amd64 libtimedate-perl all 2.3000-2 [37.5 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-date-perl all 6.02-1 [10.4 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/main amd64 libfile-listing-perl all 6.04-1 [9,774 B]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic/main amd64 libfont-afm-perl all 1.20-2 [13.2 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhtml-tagset-perl all 3.20-3 [12.1 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu bionic/main amd64 liburi-perl all 1.73-1 [77.2 kB]\n",
            "Get:38 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhtml-parser-perl amd64 3.72-3build1 [85.9 kB]\n",
            "Get:39 http://archive.ubuntu.com/ubuntu bionic/main amd64 libio-html-perl all 1.001-1 [14.9 kB]\n",
            "Get:40 http://archive.ubuntu.com/ubuntu bionic/main amd64 liblwp-mediatypes-perl all 6.02-1 [21.7 kB]\n",
            "Get:41 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-message-perl all 6.14-1 [72.1 kB]\n",
            "Get:42 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhtml-form-perl all 6.03-1 [23.5 kB]\n",
            "Get:43 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhtml-tree-perl all 5.07-1 [200 kB]\n",
            "Get:44 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhtml-format-perl all 2.12-1 [41.3 kB]\n",
            "Get:45 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-cookies-perl all 6.04-1 [17.2 kB]\n",
            "Get:46 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-daemon-perl all 6.01-1 [17.0 kB]\n",
            "Get:47 http://archive.ubuntu.com/ubuntu bionic/main amd64 libhttp-negotiate-perl all 6.00-2 [13.4 kB]\n",
            "Get:48 http://archive.ubuntu.com/ubuntu bionic/main amd64 perl-openssl-defaults amd64 3build1 [7,012 B]\n",
            "Get:49 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libnet-ssleay-perl amd64 1.84-1ubuntu0.2 [283 kB]\n",
            "Get:50 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libio-socket-ssl-perl all 2.060-3~ubuntu18.04.1 [173 kB]\n",
            "Get:51 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libkdf5-2 amd64 2.8.2-2+dfsg-1build1 [13.1 kB]\n",
            "Get:52 http://archive.ubuntu.com/ubuntu bionic/main amd64 libnet-http-perl all 6.17-1 [22.7 kB]\n",
            "Get:53 http://archive.ubuntu.com/ubuntu bionic/main amd64 libtry-tiny-perl all 0.30-1 [20.5 kB]\n",
            "Get:54 http://archive.ubuntu.com/ubuntu bionic/main amd64 libwww-robotrules-perl all 6.01-1 [14.1 kB]\n",
            "Get:55 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libwww-perl all 6.31-1ubuntu0.1 [137 kB]\n",
            "Get:56 http://archive.ubuntu.com/ubuntu bionic/main amd64 liblwp-protocol-https-perl all 6.07-2 [8,284 B]\n",
            "Get:57 http://archive.ubuntu.com/ubuntu bionic/main amd64 libnet-smtp-ssl-perl all 1.04-1 [5,948 B]\n",
            "Get:58 http://archive.ubuntu.com/ubuntu bionic/main amd64 libmailtools-perl all 2.18-1 [74.0 kB]\n",
            "Get:59 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedcrypto1 amd64 2.8.0-1 [140 kB]\n",
            "Get:60 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libmbedtls10 amd64 2.8.0-1 [66.8 kB]\n",
            "Get:61 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libncbi-vdb2 amd64 2.8.2-2+dfsg-1build1 [1,098 kB]\n",
            "Get:62 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libncbi-wvdb2 amd64 2.8.2-2+dfsg-1build1 [997 kB]\n",
            "Get:63 http://archive.ubuntu.com/ubuntu bionic/universe amd64 ncbi-entrez-direct amd64 7.40.20170928+ds-1 [810 kB]\n",
            "Get:64 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:65 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:66 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:67 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:68 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:69 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:70 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:71 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:72 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:73 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:74 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:75 http://archive.ubuntu.com/ubuntu bionic/universe amd64 sra-toolkit amd64 2.8.2-5+dfsg-1 [1,102 kB]\n",
            "Get:76 http://archive.ubuntu.com/ubuntu bionic/main amd64 libauthen-sasl-perl all 2.1600-1 [48.7 kB]\n",
            "Get:77 http://archive.ubuntu.com/ubuntu bionic/main amd64 libc-ares2 amd64 1.14.0-1 [37.1 kB]\n",
            "Get:78 http://archive.ubuntu.com/ubuntu bionic/universe amd64 med-config all 3.0.1ubuntu1 [9,404 B]\n",
            "Get:79 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 15.6 MB in 2s (7,762 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package libmagic-mgc.\n",
            "(Reading database ... 160706 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libmagic-mgc_1%3a5.32-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking libmagic-mgc (1:5.32-2ubuntu0.4) ...\n",
            "Selecting previously unselected package libmagic1:amd64.\n",
            "Preparing to unpack .../01-libmagic1_1%3a5.32-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking libmagic1:amd64 (1:5.32-2ubuntu0.4) ...\n",
            "Selecting previously unselected package netbase.\n",
            "Preparing to unpack .../02-netbase_5.4_all.deb ...\n",
            "Unpacking netbase (5.4) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../03-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package bcftools.\n",
            "Preparing to unpack .../04-bcftools_1.7-2_amd64.deb ...\n",
            "Unpacking bcftools (1.7-2) ...\n",
            "Selecting previously unselected package menu.\n",
            "Preparing to unpack .../05-menu_2.1.47ubuntu2.1_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu2.1) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../06-blends-common_0.6.100ubuntu2_all.deb ...\n",
            "Unpacking blends-common (0.6.100ubuntu2) ...\n",
            "Selecting previously unselected package bwa.\n",
            "Preparing to unpack .../07-bwa_0.7.17-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-1ubuntu0.1) ...\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "Preparing to unpack .../08-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../09-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../10-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../11-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../12-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../13-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../14-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../15-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../16-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../17-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../18-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../19-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../20-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../21-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../22-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../23-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../24-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../25-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../26-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../27-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../28-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libdata-dump-perl.\n",
            "Preparing to unpack .../29-libdata-dump-perl_1.23-1_all.deb ...\n",
            "Unpacking libdata-dump-perl (1.23-1) ...\n",
            "Selecting previously unselected package libencode-locale-perl.\n",
            "Preparing to unpack .../30-libencode-locale-perl_1.05-1_all.deb ...\n",
            "Unpacking libencode-locale-perl (1.05-1) ...\n",
            "Selecting previously unselected package libtimedate-perl.\n",
            "Preparing to unpack .../31-libtimedate-perl_2.3000-2_all.deb ...\n",
            "Unpacking libtimedate-perl (2.3000-2) ...\n",
            "Selecting previously unselected package libhttp-date-perl.\n",
            "Preparing to unpack .../32-libhttp-date-perl_6.02-1_all.deb ...\n",
            "Unpacking libhttp-date-perl (6.02-1) ...\n",
            "Selecting previously unselected package libfile-listing-perl.\n",
            "Preparing to unpack .../33-libfile-listing-perl_6.04-1_all.deb ...\n",
            "Unpacking libfile-listing-perl (6.04-1) ...\n",
            "Selecting previously unselected package libfont-afm-perl.\n",
            "Preparing to unpack .../34-libfont-afm-perl_1.20-2_all.deb ...\n",
            "Unpacking libfont-afm-perl (1.20-2) ...\n",
            "Selecting previously unselected package libhtml-tagset-perl.\n",
            "Preparing to unpack .../35-libhtml-tagset-perl_3.20-3_all.deb ...\n",
            "Unpacking libhtml-tagset-perl (3.20-3) ...\n",
            "Selecting previously unselected package liburi-perl.\n",
            "Preparing to unpack .../36-liburi-perl_1.73-1_all.deb ...\n",
            "Unpacking liburi-perl (1.73-1) ...\n",
            "Selecting previously unselected package libhtml-parser-perl.\n",
            "Preparing to unpack .../37-libhtml-parser-perl_3.72-3build1_amd64.deb ...\n",
            "Unpacking libhtml-parser-perl (3.72-3build1) ...\n",
            "Selecting previously unselected package libio-html-perl.\n",
            "Preparing to unpack .../38-libio-html-perl_1.001-1_all.deb ...\n",
            "Unpacking libio-html-perl (1.001-1) ...\n",
            "Selecting previously unselected package liblwp-mediatypes-perl.\n",
            "Preparing to unpack .../39-liblwp-mediatypes-perl_6.02-1_all.deb ...\n",
            "Unpacking liblwp-mediatypes-perl (6.02-1) ...\n",
            "Selecting previously unselected package libhttp-message-perl.\n",
            "Preparing to unpack .../40-libhttp-message-perl_6.14-1_all.deb ...\n",
            "Unpacking libhttp-message-perl (6.14-1) ...\n",
            "Selecting previously unselected package libhtml-form-perl.\n",
            "Preparing to unpack .../41-libhtml-form-perl_6.03-1_all.deb ...\n",
            "Unpacking libhtml-form-perl (6.03-1) ...\n",
            "Selecting previously unselected package libhtml-tree-perl.\n",
            "Preparing to unpack .../42-libhtml-tree-perl_5.07-1_all.deb ...\n",
            "Unpacking libhtml-tree-perl (5.07-1) ...\n",
            "Selecting previously unselected package libhtml-format-perl.\n",
            "Preparing to unpack .../43-libhtml-format-perl_2.12-1_all.deb ...\n",
            "Unpacking libhtml-format-perl (2.12-1) ...\n",
            "Selecting previously unselected package libhttp-cookies-perl.\n",
            "Preparing to unpack .../44-libhttp-cookies-perl_6.04-1_all.deb ...\n",
            "Unpacking libhttp-cookies-perl (6.04-1) ...\n",
            "Selecting previously unselected package libhttp-daemon-perl.\n",
            "Preparing to unpack .../45-libhttp-daemon-perl_6.01-1_all.deb ...\n",
            "Unpacking libhttp-daemon-perl (6.01-1) ...\n",
            "Selecting previously unselected package libhttp-negotiate-perl.\n",
            "Preparing to unpack .../46-libhttp-negotiate-perl_6.00-2_all.deb ...\n",
            "Unpacking libhttp-negotiate-perl (6.00-2) ...\n",
            "Selecting previously unselected package perl-openssl-defaults:amd64.\n",
            "Preparing to unpack .../47-perl-openssl-defaults_3build1_amd64.deb ...\n",
            "Unpacking perl-openssl-defaults:amd64 (3build1) ...\n",
            "Selecting previously unselected package libnet-ssleay-perl.\n",
            "Preparing to unpack .../48-libnet-ssleay-perl_1.84-1ubuntu0.2_amd64.deb ...\n",
            "Unpacking libnet-ssleay-perl (1.84-1ubuntu0.2) ...\n",
            "Selecting previously unselected package libio-socket-ssl-perl.\n",
            "Preparing to unpack .../49-libio-socket-ssl-perl_2.060-3~ubuntu18.04.1_all.deb ...\n",
            "Unpacking libio-socket-ssl-perl (2.060-3~ubuntu18.04.1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../50-libkdf5-2_2.8.2-2+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.8.2-2+dfsg-1build1) ...\n",
            "Selecting previously unselected package libnet-http-perl.\n",
            "Preparing to unpack .../51-libnet-http-perl_6.17-1_all.deb ...\n",
            "Unpacking libnet-http-perl (6.17-1) ...\n",
            "Selecting previously unselected package libtry-tiny-perl.\n",
            "Preparing to unpack .../52-libtry-tiny-perl_0.30-1_all.deb ...\n",
            "Unpacking libtry-tiny-perl (0.30-1) ...\n",
            "Selecting previously unselected package libwww-robotrules-perl.\n",
            "Preparing to unpack .../53-libwww-robotrules-perl_6.01-1_all.deb ...\n",
            "Unpacking libwww-robotrules-perl (6.01-1) ...\n",
            "Selecting previously unselected package libwww-perl.\n",
            "Preparing to unpack .../54-libwww-perl_6.31-1ubuntu0.1_all.deb ...\n",
            "Unpacking libwww-perl (6.31-1ubuntu0.1) ...\n",
            "Selecting previously unselected package liblwp-protocol-https-perl.\n",
            "Preparing to unpack .../55-liblwp-protocol-https-perl_6.07-2_all.deb ...\n",
            "Unpacking liblwp-protocol-https-perl (6.07-2) ...\n",
            "Selecting previously unselected package libnet-smtp-ssl-perl.\n",
            "Preparing to unpack .../56-libnet-smtp-ssl-perl_1.04-1_all.deb ...\n",
            "Unpacking libnet-smtp-ssl-perl (1.04-1) ...\n",
            "Selecting previously unselected package libmailtools-perl.\n",
            "Preparing to unpack .../57-libmailtools-perl_2.18-1_all.deb ...\n",
            "Unpacking libmailtools-perl (2.18-1) ...\n",
            "Selecting previously unselected package libmbedcrypto1:amd64.\n",
            "Preparing to unpack .../58-libmbedcrypto1_2.8.0-1_amd64.deb ...\n",
            "Unpacking libmbedcrypto1:amd64 (2.8.0-1) ...\n",
            "Selecting previously unselected package libmbedtls10:amd64.\n",
            "Preparing to unpack .../59-libmbedtls10_2.8.0-1_amd64.deb ...\n",
            "Unpacking libmbedtls10:amd64 (2.8.0-1) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../60-libncbi-vdb2_2.8.2-2+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../61-libncbi-wvdb2_2.8.2-2+dfsg-1build1_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Selecting previously unselected package ncbi-entrez-direct.\n",
            "Preparing to unpack .../62-ncbi-entrez-direct_7.40.20170928+ds-1_amd64.deb ...\n",
            "Adding 'diversion of /usr/bin/efetch to /usr/bin/efetch.acedb by ncbi-entrez-direct'\n",
            "Adding 'diversion of /usr/share/man/man1/efetch.1.gz to /usr/share/man/man1/efetch.acedb.1.gz by ncbi-entrez-direct'\n",
            "Adding 'diversion of /usr/bin/einfo to /usr/bin/einfo.epub by ncbi-entrez-direct'\n",
            "Adding 'diversion of /usr/share/man/man1/einfo.1.gz to /usr/share/man/man1/einfo.epub.1.gz by ncbi-entrez-direct'\n",
            "Unpacking ncbi-entrez-direct (7.40.20170928+ds-1) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../63-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../64-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../65-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../66-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../67-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../68-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../69-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../70-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../71-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../72-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../73-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../74-sra-toolkit_2.8.2-5+dfsg-1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.8.2-5+dfsg-1) ...\n",
            "Selecting previously unselected package libauthen-sasl-perl.\n",
            "Preparing to unpack .../75-libauthen-sasl-perl_2.1600-1_all.deb ...\n",
            "Unpacking libauthen-sasl-perl (2.1600-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../76-libc-ares2_1.14.0-1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../77-med-config_3.0.1ubuntu1_all.deb ...\n",
            "Unpacking med-config (3.0.1ubuntu1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../78-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up libhtml-tagset-perl (3.20-3) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up libtry-tiny-perl (0.30-1) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up bwa (0.7.17-1ubuntu0.1) ...\n",
            "Setting up libfont-afm-perl (1.20-2) ...\n",
            "Setting up libencode-locale-perl (1.05-1) ...\n",
            "Setting up libtimedate-perl (2.3000-2) ...\n",
            "Setting up perl-openssl-defaults:amd64 (3build1) ...\n",
            "Setting up libmbedcrypto1:amd64 (2.8.0-1) ...\n",
            "Setting up libio-html-perl (1.001-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up menu (2.1.47ubuntu2.1) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up libkdf5-2 (2.8.2-2+dfsg-1build1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up libmagic-mgc (1:5.32-2ubuntu0.4) ...\n",
            "Setting up libmagic1:amd64 (1:5.32-2ubuntu0.4) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up liblwp-mediatypes-perl (6.02-1) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up liburi-perl (1.73-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up libdata-dump-perl (1.23-1) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libhtml-parser-perl (3.72-3build1) ...\n",
            "Setting up libncbi-wvdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Setting up libnet-http-perl (6.17-1) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1) ...\n",
            "Setting up libwww-robotrules-perl (6.01-1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up libauthen-sasl-perl (2.1600-1) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up netbase (5.4) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up libhttp-date-perl (6.02-1) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up libmbedtls10:amd64 (2.8.0-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up libnet-ssleay-perl (1.84-1ubuntu0.2) ...\n",
            "Setting up blends-common (0.6.100ubuntu2) ...\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up libncbi-vdb2 (2.8.2-2+dfsg-1build1) ...\n",
            "Setting up sra-toolkit (2.8.2-5+dfsg-1) ...\n",
            "Setting up libio-socket-ssl-perl (2.060-3~ubuntu18.04.1) ...\n",
            "Setting up libhtml-tree-perl (5.07-1) ...\n",
            "Setting up med-config (3.0.1ubuntu1) ...\n",
            "Adding group `med' (GID 109) ...\n",
            "Done.\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up libfile-listing-perl (6.04-1) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up libhttp-message-perl (6.14-1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up bcftools (1.7-2) ...\n",
            "Setting up libhttp-negotiate-perl (6.00-2) ...\n",
            "Setting up libnet-smtp-ssl-perl (1.04-1) ...\n",
            "Setting up libhtml-format-perl (2.12-1) ...\n",
            "Setting up libhttp-cookies-perl (6.04-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up libhttp-daemon-perl (6.01-1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Setting up libhtml-form-perl (6.03-1) ...\n",
            "Setting up libmailtools-perl (2.18-1) ...\n",
            "Setting up libwww-perl (6.31-1ubuntu0.1) ...\n",
            "Setting up ncbi-entrez-direct (7.40.20170928+ds-1) ...\n",
            "Setting up liblwp-protocol-https-perl (6.07-2) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.2) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Processing triggers for menu (2.1.47ubuntu2.1) ...\n",
            "Collecting snptoolkit\n",
            "  Downloading https://files.pythonhosted.org/packages/de/52/795b786d0251f43273b1e0b4fed6dd08af27f81c3a097274359264a5c846/snptoolkit-2.3.0.tar.gz\n",
            "Collecting pysam\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/20/85/335857b9888f6d9a13b03a8f21b0a6228b180c361631d9d70e7be3e22163/pysam-0.16.0.1-cp37-cp37m-manylinux1_x86_64.whl (9.9MB)\n",
            "\u001b[K     |████████████████████████████████| 9.9MB 19.6MB/s \n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.7/dist-packages (from snptoolkit) (1.1.5)\n",
            "Collecting Biopython\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/3a/cd/0098eaff841850c01da928c7f509b72fd3e1f51d77b772e24de9e2312471/biopython-1.78-cp37-cp37m-manylinux1_x86_64.whl (2.3MB)\n",
            "\u001b[K     |████████████████████████████████| 2.3MB 48.9MB/s \n",
            "\u001b[?25hCollecting dash\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/d4/50/e7c2830168db186f84b7de2988543e974433a6cdb0a0b23d51c781e2b2ab/dash-1.20.0.tar.gz (77kB)\n",
            "\u001b[K     |████████████████████████████████| 81kB 10.8MB/s \n",
            "\u001b[?25hCollecting dash-daq\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/16/c1/9c6adfc3646990965a5b2ca073f579cc9c09d15553fcfcd83aefd1255494/dash_daq-0.5.0.tar.gz (642kB)\n",
            "\u001b[K     |████████████████████████████████| 645kB 47.1MB/s \n",
            "\u001b[?25hRequirement already satisfied: plotly in /usr/local/lib/python3.7/dist-packages (from snptoolkit) (4.4.1)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.7/dist-packages (from snptoolkit) (0.22.2.post1)\n",
            "Collecting umap-learn==0.4.6\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/ac/21/e1eb2eb1c624a84f4a23237adb974e94ff7371ee8c178d246194af83fb80/umap-learn-0.4.6.tar.gz (69kB)\n",
            "\u001b[K     |████████████████████████████████| 71kB 8.9MB/s \n",
            "\u001b[?25hRequirement already satisfied: llvmlite==0.34.0 in /usr/local/lib/python3.7/dist-packages (from snptoolkit) (0.34.0)\n",
            "Requirement already satisfied: numba==0.51.2 in /usr/local/lib/python3.7/dist-packages (from snptoolkit) (0.51.2)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.7/dist-packages (from snptoolkit) (4.41.1)\n",
            "Collecting coloredlogs\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/84/a6/837dbf6eac344cb74f0ba86b79e8180855570af3e26bcb1ea5f650cf944c/coloredlogs-15.0-py2.py3-none-any.whl (45kB)\n",
            "\u001b[K     |████████████████████████████████| 51kB 7.5MB/s \n",
            "\u001b[?25hRequirement already satisfied: python-dateutil>=2.7.3 in /usr/local/lib/python3.7/dist-packages (from pandas->snptoolkit) (2.8.1)\n",
            "Requirement already satisfied: numpy>=1.15.4 in /usr/local/lib/python3.7/dist-packages (from pandas->snptoolkit) (1.19.5)\n",
            "Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.7/dist-packages (from pandas->snptoolkit) (2018.9)\n",
            "Requirement already satisfied: Flask>=1.0.4 in /usr/local/lib/python3.7/dist-packages (from dash->snptoolkit) (1.1.2)\n",
            "Collecting flask-compress\n",
            "  Downloading https://files.pythonhosted.org/packages/c6/d5/69b13600230d24310b98a52da561113fc01a5c17acf77152761eef3e50f1/Flask_Compress-1.9.0-py3-none-any.whl\n",
            "Collecting dash_renderer==1.9.1\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/5f/d3/d661a68b4ce71498d5c0c79617bce3d5fc884d4448c698f77c2247cd1b46/dash_renderer-1.9.1.tar.gz (1.0MB)\n",
            "\u001b[K     |████████████████████████████████| 1.0MB 38.1MB/s \n",
            "\u001b[?25hCollecting dash-core-components==1.16.0\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/f4/79/434e14d77dbf82b27f98df077936a424de254059d47152b0445b7116e97e/dash_core_components-1.16.0.tar.gz (3.5MB)\n",
            "\u001b[K     |████████████████████████████████| 3.5MB 34.0MB/s \n",
            "\u001b[?25hCollecting dash-html-components==1.1.3\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/8f/ca/7eeca4cae4e15cc6f7d0bc41a46368dd3fcdddc72aaa5bac61a8b92d541c/dash_html_components-1.1.3.tar.gz (82kB)\n",
            "\u001b[K     |████████████████████████████████| 92kB 12.2MB/s \n",
            "\u001b[?25hCollecting dash-table==4.11.3\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/97/f7/f4969a926f20a55d3e5970d01b85ff9ad510dba32de189e72dd8f4992740/dash_table-4.11.3.tar.gz (1.8MB)\n",
            "\u001b[K     |████████████████████████████████| 1.8MB 41.7MB/s \n",
            "\u001b[?25hRequirement already satisfied: future in /usr/local/lib/python3.7/dist-packages (from dash->snptoolkit) (0.16.0)\n",
            "Requirement already satisfied: retrying>=1.3.3 in /usr/local/lib/python3.7/dist-packages (from plotly->snptoolkit) (1.3.3)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from plotly->snptoolkit) (1.15.0)\n",
            "Requirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.7/dist-packages (from scikit-learn->snptoolkit) (1.0.1)\n",
            "Requirement already satisfied: scipy>=0.17.0 in /usr/local/lib/python3.7/dist-packages (from scikit-learn->snptoolkit) (1.4.1)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from numba==0.51.2->snptoolkit) (56.1.0)\n",
            "Collecting humanfriendly>=9.1\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/93/66/363d01a81da2108a5cf446daf619779f06d49a0c4426dd02b40734f10e2f/humanfriendly-9.1-py2.py3-none-any.whl (86kB)\n",
            "\u001b[K     |████████████████████████████████| 92kB 10.5MB/s \n",
            "\u001b[?25hRequirement already satisfied: click>=5.1 in /usr/local/lib/python3.7/dist-packages (from Flask>=1.0.4->dash->snptoolkit) (8.0.0)\n",
            "Requirement already satisfied: Werkzeug>=0.15 in /usr/local/lib/python3.7/dist-packages (from Flask>=1.0.4->dash->snptoolkit) (2.0.0)\n",
            "Requirement already satisfied: Jinja2>=2.10.1 in /usr/local/lib/python3.7/dist-packages (from Flask>=1.0.4->dash->snptoolkit) (2.11.3)\n",
            "Requirement already satisfied: itsdangerous>=0.24 in /usr/local/lib/python3.7/dist-packages (from Flask>=1.0.4->dash->snptoolkit) (2.0.0)\n",
            "Collecting brotli\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/15/ea/5bd575511b37bbd1c794606a0a621e6feff8e96b7dd007a86a5d218b2d94/Brotli-1.0.9-cp37-cp37m-manylinux1_x86_64.whl (357kB)\n",
            "\u001b[K     |████████████████████████████████| 358kB 43.9MB/s \n",
            "\u001b[?25hRequirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.7/dist-packages (from Jinja2>=2.10.1->Flask>=1.0.4->dash->snptoolkit) (2.0.0)\n",
            "Building wheels for collected packages: snptoolkit, dash, dash-daq, umap-learn, dash-renderer, dash-core-components, dash-html-components, dash-table\n",
            "  Building wheel for snptoolkit (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for snptoolkit: filename=snptoolkit-2.3.0-cp37-none-any.whl size=30850 sha256=5d350e5e51535cac9ff65a7b62154f584df124a0f0283fdaceb47c1ea91f06f8\n",
            "  Stored in directory: /root/.cache/pip/wheels/33/6a/4c/af428b0ad0654eb7de9cc8e6915844da27ede7eb0c9282d7c9\n",
            "  Building wheel for dash (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for dash: filename=dash-1.20.0-cp37-none-any.whl size=85834 sha256=2226a76b4348543ef0cacbee7149fc54bb45a018fb94dc277541a59fcf662445\n",
            "  Stored in directory: /root/.cache/pip/wheels/e6/33/41/ce361d0d1da277d2350f815b10f2ab1e331592d93ef2b07e29\n",
            "  Building wheel for dash-daq (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for dash-daq: filename=dash_daq-0.5.0-cp37-none-any.whl size=669704 sha256=45a465fe49d12769bd000e1d5e20eb2d103603ce9e1709fb4634529126ab01b4\n",
            "  Stored in directory: /root/.cache/pip/wheels/1c/9c/f7/84731716eaa9fc952cdbdd1f87df3ca9b8805c317c609efcd6\n",
            "  Building wheel for umap-learn (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for umap-learn: filename=umap_learn-0.4.6-cp37-none-any.whl size=67950 sha256=5df4ef677dbe883b2aa68a56a3a6df6eaf9cf867e9bbaaca2345af2b07793989\n",
            "  Stored in directory: /root/.cache/pip/wheels/7d/1d/03/34aade9a9b97acddb8e93654eb856dadbf0964406eef8b96e2\n",
            "  Building wheel for dash-renderer (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for dash-renderer: filename=dash_renderer-1.9.1-cp37-none-any.whl size=1014875 sha256=31ecfec4c1855d8e7be52f8a4ab0dd68adee5b7232bf0ab89f8b5e536bd9e514\n",
            "  Stored in directory: /root/.cache/pip/wheels/03/a9/c5/dd5815c601b0ede164c223ffd7bafebde716ca57de06ef8aec\n",
            "  Building wheel for dash-core-components (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for dash-core-components: filename=dash_core_components-1.16.0-cp37-none-any.whl size=3540993 sha256=9564ba6a15ba983927b1ac5b3217d938c8d294a9b43fa81dd49b27376958d390\n",
            "  Stored in directory: /root/.cache/pip/wheels/86/1e/8c/e87ebba30b73c20dcd641224274febc983af88ed0fd7712a07\n",
            "  Building wheel for dash-html-components (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for dash-html-components: filename=dash_html_components-1.1.3-cp37-none-any.whl size=319471 sha256=b7927d8df0308deb8ba33553e1c5a17c6cfe81b0465786e2a6b7eb68a43b4322\n",
            "  Stored in directory: /root/.cache/pip/wheels/07/f9/6c/f9b73a6ae1b7f347a92dc240293cebc267b370ba2a80added2\n",
            "  Building wheel for dash-table (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for dash-table: filename=dash_table-4.11.3-cp37-none-any.whl size=1827619 sha256=2e8165db12b966e6dd594aad6ec09543eef4c2c613a0f12e08bb55df7180a4af\n",
            "  Stored in directory: /root/.cache/pip/wheels/51/9b/89/374be1bc250e28a15edbd657cf364f163e392ba5257bfd0e87\n",
            "Successfully built snptoolkit dash dash-daq umap-learn dash-renderer dash-core-components dash-html-components dash-table\n",
            "Installing collected packages: pysam, Biopython, brotli, flask-compress, dash-renderer, dash-core-components, dash-html-components, dash-table, dash, dash-daq, umap-learn, humanfriendly, coloredlogs, snptoolkit\n",
            "Successfully installed Biopython-1.78 brotli-1.0.9 coloredlogs-15.0 dash-1.20.0 dash-core-components-1.16.0 dash-daq-0.5.0 dash-html-components-1.1.3 dash-renderer-1.9.1 dash-table-4.11.3 flask-compress-1.9.0 humanfriendly-9.1 pysam-0.16.0.1 snptoolkit-2.3.0 umap-learn-0.4.6\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "m2xWlKpie4C5"
      },
      "source": [
        "#### Baixe a sequência e anotação do genoma do vírus SARS-Cov-2 isolado Wuhan-Hu-1 do GenBank (GenBank: NC_045512.2), que servirá como referência. "
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6cVVYlEZMcV2"
      },
      "source": [
        "!efetch -db nucleotide -id NC_045512.2 -format fasta > NC_045512.2.fasta\n",
        "!efetch -db nucleotide -id NC_045512.2 -format gbwithparts > NC_045512.2.gb"
      ],
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "p-NNezYyNcv0",
        "outputId": "1d214e36-0f44-49d2-dd83-7eefec353812"
      },
      "source": [
        "!head -100 NC_045512.2.fasta"
      ],
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            ">NC_045512.2 Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n",
            "ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAA\n",
            "CGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAAC\n",
            "TAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTG\n",
            "TTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTC\n",
            "CCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTAC\n",
            "GTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGG\n",
            "CTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGAT\n",
            "GCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTC\n",
            "GTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCT\n",
            "TCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTA\n",
            "GGCGACGAGCTTGGCACTGATCCTTATGAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTG\n",
            "TTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGG\n",
            "CCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTCATGCACTTTG\n",
            "TCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTG\n",
            "CTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAA\n",
            "ATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCCTTAAATTCCATAATCAAGACTATTCAA\n",
            "CCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTTATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCAC\n",
            "CAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCA\n",
            "GACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACT\n",
            "ACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAATTTATTGTCCAGCATGTCACAATTCAGAAGTAG\n",
            "GACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCG\n",
            "CACTATTGCCTTTGGAGGCTGTGTGTTCTCTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCA\n",
            "CGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACA\n",
            "ACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGA\n",
            "GATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGAT\n",
            "TATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAAAAAG\n",
            "GTGCCTGGAATATTGGTGAACAGAAATCAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCG\n",
            "TGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCC\n",
            "GCTATAACAATACTAGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTG\n",
            "ATTTGGCTACTAACAATCTAGTTGTAATGGCCTACATTACAGGTGGTGTTGTTCAGTTGACTTCGCAGTG\n",
            "GCTAACTAACATCTTTGGCACTGTTTATGAAAAACTCAAACCCGTCCTTGATTGGCTTGAAGAGAAGTTT\n",
            "AAGGAAGGTGTAGAGTTTCTTAGAGACGGTTGGGAAATTGTTAAATTTATCTCAACCTGTGCTTGTGAAA\n",
            "TTGTCGGTGGACAAATTGTCACCTGTGCAAAGGAAATTAAGGAGAGTGTTCAGACATTCTTTAAGCTTGT\n",
            "AAATAAATTTTTGGCTTTGTGTGCTGACTCTATCATTATTGGTGGAGCTAAACTTAAAGCCTTGAATTTA\n",
            "GGTGAAACATTTGTCACGCACTCAAAGGGATTGTACAGAAAGTGTGTTAAATCCAGAGAAGAAACTGGCC\n",
            "TACTCATGCCTCTAAAAGCCCCAAAAGAAATTATCTTCTTAGAGGGAGAAACACTTCCCACAGAAGTGTT\n",
            "AACAGAGGAAGTTGTCTTGAAAACTGGTGATTTACAACCATTAGAACAACCTACTAGTGAAGCTGTTGAA\n",
            "GCTCCATTGGTTGGTACACCAGTTTGTATTAACGGGCTTATGTTGCTCGAAATCAAAGACACAGAAAAGT\n",
            "ACTGTGCCCTTGCACCTAATATGATGGTAACAAACAATACCTTCACACTCAAAGGCGGTGCACCAACAAA\n",
            "GGTTACTTTTGGTGATGACACTGTGATAGAAGTGCAAGGTTACAAGAGTGTGAATATCACTTTTGAACTT\n",
            "GATGAAAGGATTGATAAAGTACTTAATGAGAAGTGCTCTGCCTATACAGTTGAACTCGGTACAGAAGTAA\n",
            "ATGAGTTCGCCTGTGTTGTGGCAGATGCTGTCATAAAAACTTTGCAACCAGTATCTGAATTACTTACACC\n",
            "ACTGGGCATTGATTTAGATGAGTGGAGTATGGCTACATACTACTTATTTGATGAGTCTGGTGAGTTTAAA\n",
            "TTGGCTTCACATATGTATTGTTCTTTCTACCCTCCAGATGAGGATGAAGAAGAAGGTGATTGTGAAGAAG\n",
            "AAGAGTTTGAGCCATCAACTCAATATGAGTATGGTACTGAAGATGATTACCAAGGTAAACCTTTGGAATT\n",
            "TGGTGCCACTTCTGCTGCTCTTCAACCTGAAGAAGAGCAAGAAGAAGATTGGTTAGATGATGATAGTCAA\n",
            "CAAACTGTTGGTCAACAAGACGGCAGTGAGGACAATCAGACAACTACTATTCAAACAATTGTTGAGGTTC\n",
            "AACCTCAATTAGAGATGGAACTTACACCAGTTGTTCAGACTATTGAAGTGAATAGTTTTAGTGGTTATTT\n",
            "AAAACTTACTGACAATGTATACATTAAAAATGCAGACATTGTGGAAGAAGCTAAAAAGGTAAAACCAACA\n",
            "GTGGTTGTTAATGCAGCCAATGTTTACCTTAAACATGGAGGAGGTGTTGCAGGAGCCTTAAATAAGGCTA\n",
            "CTAACAATGCCATGCAAGTTGAATCTGATGATTACATAGCTACTAATGGACCACTTAAAGTGGGTGGTAG\n",
            "TTGTGTTTTAAGCGGACACAATCTTGCTAAACACTGTCTTCATGTTGTCGGCCCAAATGTTAACAAAGGT\n",
            "GAAGACATTCAACTTCTTAAGAGTGCTTATGAAAATTTTAATCAGCACGAAGTTCTACTTGCACCATTAT\n",
            "TATCAGCTGGTATTTTTGGTGCTGACCCTATACATTCTTTAAGAGTTTGTGTAGATACTGTTCGCACAAA\n",
            "TGTCTACTTAGCTGTCTTTGATAAAAATCTCTATGACAAACTTGTTTCAAGCTTTTTGGAAATGAAGAGT\n",
            "GAAAAGCAAGTTGAACAAAAGATCGCTGAGATTCCTAAAGAGGAAGTTAAGCCATTTATAACTGAAAGTA\n",
            "AACCTTCAGTTGAACAGAGAAAACAAGATGATAAGAAAATCAAAGCTTGTGTTGAAGAAGTTACAACAAC\n",
            "TCTGGAAGAAACTAAGTTCCTCACAGAAAACTTGTTACTTTATATTGACATTAATGGCAATCTTCATCCA\n",
            "GATTCTGCCACTCTTGTTAGTGACATTGACATCACTTTCTTAAAGAAAGATGCTCCATATATAGTGGGTG\n",
            "ATGTTGTTCAAGAGGGTGTTTTAACTGCTGTGGTTATACCTACTAAAAAGGCTGGTGGCACTACTGAAAT\n",
            "GCTAGCGAAAGCTTTGAGAAAAGTGCCAACAGACAATTATATAACCACTTACCCGGGTCAGGGTTTAAAT\n",
            "GGTTACACTGTAGAGGAGGCAAAGACAGTGCTTAAAAAGTGTAAAAGTGCCTTTTACATTCTACCATCTA\n",
            "TTATCTCTAATGAGAAGCAAGAAATTCTTGGAACTGTTTCTTGGAATTTGCGAGAAATGCTTGCACATGC\n",
            "AGAAGAAACACGCAAATTAATGCCTGTCTGTGTGGAAACTAAAGCCATAGTTTCAACTATACAGCGTAAA\n",
            "TATAAGGGTATTAAAATACAAGAGGGTGTGGTTGATTATGGTGCTAGATTTTACTTTTACACCAGTAAAA\n",
            "CAACTGTAGCGTCACTTATCAACACACTTAACGATCTAAATGAAACTCTTGTTACAATGCCACTTGGCTA\n",
            "TGTAACACATGGCTTAAATTTGGAAGAAGCTGCTCGGTATATGAGATCTCTCAAAGTGCCAGCTACAGTT\n",
            "TCTGTTTCTTCACCTGATGCTGTTACAGCGTATAATGGTTATCTTACTTCTTCTTCTAAAACACCTGAAG\n",
            "AACATTTTATTGAAACCATCTCACTTGCTGGTTCCTATAAAGATTGGTCCTATTCTGGACAATCTACACA\n",
            "ACTAGGTATAGAATTTCTTAAGAGAGGTGATAAAAGTGTATATTACACTAGTAATCCTACCACATTCCAC\n",
            "CTAGATGGTGAAGTTATCACCTTTGACAATCTTAAGACACTTCTTTCTTTGAGAGAAGTGAGGACTATTA\n",
            "AGGTGTTTACAACAGTAGACAACATTAACCTCCACACGCAAGTTGTGGACATGTCAATGACATATGGACA\n",
            "ACAGTTTGGTCCAACTTATTTGGATGGAGCTGATGTTACTAAAATAAAACCTCATAATTCACATGAAGGT\n",
            "AAAACATTTTATGTTTTACCTAATGATGACACTCTACGTGTTGAGGCTTTTGAGTACTACCACACAACTG\n",
            "ATCCTAGTTTTCTGGGTAGGTACATGTCAGCATTAAATCACACTAAAAAGTGGAAATACCCACAAGTTAA\n",
            "TGGTTTAACTTCTATTAAATGGGCAGATAACAACTGTTATCTTGCCACTGCATTGTTAACACTCCAACAA\n",
            "ATAGAGTTGAAGTTTAATCCACCTGCTCTACAAGATGCTTATTACAGAGCAAGGGCTGGTGAAGCTGCTA\n",
            "ACTTTTGTGCACTTATCTTAGCCTACTGTAATAAGACAGTAGGTGAGTTAGGTGATGTTAGAGAAACAAT\n",
            "GAGTTACTTGTTTCAACATGCCAATTTAGATTCTTGCAAAAGAGTCTTGAACGTGGTGTGTAAAACTTGT\n",
            "GGACAACAGCAGACAACCCTTAAGGGTGTAGAAGCTGTTATGTACATGGGCACACTTTCTTATGAACAAT\n",
            "TTAAGAAAGGTGTTCAGATACCTTGTACGTGTGGTAAACAAGCTACAAAATATCTAGTACAACAGGAGTC\n",
            "ACCTTTTGTTATGATGTCAGCACCACCTGCTCAGTATGAACTTAAGCATGGTACATTTACTTGTGCTAGT\n",
            "GAGTACACTGGTAATTACCAGTGTGGTCACTATAAACATATAACTTCTAAAGAAACTTTGTATTGCATAG\n",
            "ACGGTGCTTTACTTACAAAGTCCTCAGAATACAAAGGTCCTATTACGGATGTTTTCTACAAAGAAAACAG\n",
            "TTACACAACAACCATAAAACCAGTTACTTATAAATTGGATGGTGTTGTTTGTACAGAAATTGACCCTAAG\n",
            "TTGGACAATTATTATAAGAAAGACAATTCTTATTTCACAGAGCAACCAATTGATCTTGTACCAAACCAAC\n",
            "CATATCCAAACGCAAGCTTCGATAATTTTAAGTTTGTATGTGATAATATCAAATTTGCTGATGATTTAAA\n",
            "CCAGTTAACTGGTTATAAGAAACCTGCTTCAAGAGAGCTTAAAGTTACATTTTTCCCTGACTTAAATGGT\n",
            "GATGTGGTGGCTATTGATTATAAACACTACACACCCTCTTTTAAGAAAGGAGCTAAATTGTTACATAAAC\n",
            "CTATTGTTTGGCATGTTAACAATGCAACTAATAAAGCCACGTATAAACCAAATACCTGGTGTATACGTTG\n",
            "TCTTTGGAGCACAAAACCAGTTGAAACATCAAATTCGTTTGATGTACTGAAGTCAGAGGACGCGCAGGGA\n",
            "ATGGATAATCTTGCCTGCGAAGATCTAAAACCAGTCTCTGAAGAAGTAGTGGAAAATCCTACCATACAGA\n",
            "AAGACGTTCTTGAGTGTAATGTGAAAACTACCGAAGTTGTAGGAGACATTATACTTAAACCAGCAAATAA\n",
            "TAGTTTAAAAATTACAGAAGAGGTTGGCCACACAGATCTAATGGCTGCTTATGTAGACAATTCTAGTCTT\n",
            "ACTATTAAGAAACCTAATGAATTATCTAGAGTATTAGGTTTGAAAACCCTTGCTACTCATGGTTTAGCTG\n",
            "CTGTTAATAGTGTCCCTTGGGATACTATAGCTAATTATGCTAAGCCTTTTCTTAACAAAGTTGTTAGTAC\n",
            "AACTACTAACATAGTTACACGGTGTTTAAACCGTGTTTGTACTAATTATATGCCTTATTTCTTTACTTTA\n",
            "TTGCTACAATTGTGTACTTTTACTAGAAGTACAAATTCTAGAATTAAAGCATCTATGCCGACTACTATAG\n",
            "CAAAGAATACTGTTAAGAGTGTCGGTAAATTTTGTCTAGAGGCTTCATTTAATTATTTGAAGTCACCTAA\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GWW0EaoENpPC",
        "outputId": "804070a1-2610-40fb-ea92-37e8ab96484a"
      },
      "source": [
        "!head -100 NC_045512.2.gb"
      ],
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "LOCUS       NC_045512              29903 bp ss-RNA     linear   VRL 18-JUL-2020\n",
            "DEFINITION  Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1,\n",
            "            complete genome.\n",
            "ACCESSION   NC_045512\n",
            "VERSION     NC_045512.2\n",
            "DBLINK      BioProject: PRJNA485481\n",
            "KEYWORDS    RefSeq.\n",
            "SOURCE      Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)\n",
            "  ORGANISM  Severe acute respiratory syndrome coronavirus 2\n",
            "            Viruses; Riboviria; Orthornavirae; Pisuviricota; Pisoniviricetes;\n",
            "            Nidovirales; Cornidovirineae; Coronaviridae; Orthocoronavirinae;\n",
            "            Betacoronavirus; Sarbecovirus.\n",
            "REFERENCE   1  (bases 1 to 29903)\n",
            "  AUTHORS   Wu,F., Zhao,S., Yu,B., Chen,Y.M., Wang,W., Song,Z.G., Hu,Y.,\n",
            "            Tao,Z.W., Tian,J.H., Pei,Y.Y., Yuan,M.L., Zhang,Y.L., Dai,F.H.,\n",
            "            Liu,Y., Wang,Q.M., Zheng,J.J., Xu,L., Holmes,E.C. and Zhang,Y.Z.\n",
            "  TITLE     A new coronavirus associated with human respiratory disease in\n",
            "            China\n",
            "  JOURNAL   Nature 579 (7798), 265-269 (2020)\n",
            "   PUBMED   32015508\n",
            "  REMARK    Erratum:[Nature. 2020 Apr;580(7803):E7. PMID: 32296181]\n",
            "REFERENCE   2  (bases 13476 to 13503)\n",
            "  AUTHORS   Baranov,P.V., Henderson,C.M., Anderson,C.B., Gesteland,R.F.,\n",
            "            Atkins,J.F. and Howard,M.T.\n",
            "  TITLE     Programmed ribosomal frameshifting in decoding the SARS-CoV genome\n",
            "  JOURNAL   Virology 332 (2), 498-510 (2005)\n",
            "   PUBMED   15680415\n",
            "REFERENCE   3  (bases 29728 to 29768)\n",
            "  AUTHORS   Robertson,M.P., Igel,H., Baertsch,R., Haussler,D., Ares,M. Jr. and\n",
            "            Scott,W.G.\n",
            "  TITLE     The structure of a rigorously conserved RNA element within the SARS\n",
            "            virus genome\n",
            "  JOURNAL   PLoS Biol. 3 (1), e5 (2005)\n",
            "   PUBMED   15630477\n",
            "REFERENCE   4  (bases 29609 to 29657)\n",
            "  AUTHORS   Williams,G.D., Chang,R.Y. and Brian,D.A.\n",
            "  TITLE     A phylogenetically conserved hairpin-type 3' untranslated region\n",
            "            pseudoknot functions in coronavirus RNA replication\n",
            "  JOURNAL   J. Virol. 73 (10), 8349-8355 (1999)\n",
            "   PUBMED   10482585\n",
            "REFERENCE   5  (bases 1 to 29903)\n",
            "  CONSRTM   NCBI Genome Project\n",
            "  TITLE     Direct Submission\n",
            "  JOURNAL   Submitted (17-JAN-2020) National Center for Biotechnology\n",
            "            Information, NIH, Bethesda, MD 20894, USA\n",
            "REFERENCE   6  (bases 1 to 29903)\n",
            "  AUTHORS   Wu,F., Zhao,S., Yu,B., Chen,Y.-M., Wang,W., Hu,Y., Song,Z.-G.,\n",
            "            Tao,Z.-W., Tian,J.-H., Pei,Y.-Y., Yuan,M.L., Zhang,Y.-L.,\n",
            "            Dai,F.-H., Liu,Y., Wang,Q.-M., Zheng,J.-J., Xu,L., Holmes,E.C. and\n",
            "            Zhang,Y.-Z.\n",
            "  TITLE     Direct Submission\n",
            "  JOURNAL   Submitted (05-JAN-2020) Shanghai Public Health Clinical Center &\n",
            "            School of Public Health, Fudan University, Shanghai, China\n",
            "COMMENT     REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
            "            reference sequence is identical to MN908947.\n",
            "            On Jan 17, 2020 this sequence version replaced NC_045512.1.\n",
            "            Annotation was added using homology to SARSr-CoV NC_004718.3. ###\n",
            "            Formerly called 'Wuhan seafood market pneumonia virus.' If you have\n",
            "            questions or suggestions, please email us at info@ncbi.nlm.nih.gov\n",
            "            and include the accession number NC_045512.### Protein structures\n",
            "            can be found at\n",
            "            https://www.ncbi.nlm.nih.gov/structure/?term=sars-cov-2.### Find\n",
            "            all other Severe acute respiratory syndrome coronavirus 2\n",
            "            (SARS-CoV-2) sequences at\n",
            "            https://www.ncbi.nlm.nih.gov/genbank/sars-cov-2-seqs/\n",
            "            \n",
            "            ##Assembly-Data-START##\n",
            "            Assembly Method       :: Megahit v. V1.1.3\n",
            "            Sequencing Technology :: Illumina\n",
            "            ##Assembly-Data-END##\n",
            "            COMPLETENESS: full length.\n",
            "FEATURES             Location/Qualifiers\n",
            "     source          1..29903\n",
            "                     /organism=\"Severe acute respiratory syndrome coronavirus\n",
            "                     2\"\n",
            "                     /mol_type=\"genomic RNA\"\n",
            "                     /isolate=\"Wuhan-Hu-1\"\n",
            "                     /host=\"Homo sapiens\"\n",
            "                     /db_xref=\"taxon:2697049\"\n",
            "                     /country=\"China\"\n",
            "                     /collection_date=\"Dec-2019\"\n",
            "     5'UTR           1..265\n",
            "     gene            266..21555\n",
            "                     /gene=\"ORF1ab\"\n",
            "                     /locus_tag=\"GU280_gp01\"\n",
            "                     /db_xref=\"GeneID:43740578\"\n",
            "     CDS             join(266..13468,13468..21555)\n",
            "                     /gene=\"ORF1ab\"\n",
            "                     /locus_tag=\"GU280_gp01\"\n",
            "                     /ribosomal_slippage\n",
            "                     /note=\"pp1ab; translated by -1 ribosomal frameshift\"\n",
            "                     /codon_start=1\n",
            "                     /product=\"ORF1ab polyprotein\"\n",
            "                     /protein_id=\"YP_009724389.1\"\n",
            "                     /db_xref=\"GeneID:43740578\"\n",
            "                     /translation=\"MESLVPGFNEKTHVQLSLPVLQVRDVLVRGFGDSVEEVLSEARQ\n",
            "                     HLKDGTCGLVEVEKGVLPQLEQPYVFIKRSDARTAPHGHVMVELVAELEGIQYGRSGE\n",
            "                     TLGVLVPHVGEIPVAYRKVLLRKNGNKGAGGHSYGADLKSFDLGDELGTDPYEDFQEN\n",
            "                     WNTKHSSGVTRELMRELNGGAYTRYVDNNFCGPDGYPLECIKDLLARAGKASCTLSEQ\n",
            "                     LDFIDTKRGVYCCREHEHEIAWYTERSEKSYELQTPFEIKLAKKFDTFNGECPNFVFP\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "surVU5krfPWx"
      },
      "source": [
        "#### Baixe os dados brutos de sequenciamento de um novo isolado de SARS-Cov-2 obtido na Espanha (SRA: ERR5908782)."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "m_OE4sXJNtwq",
        "outputId": "53d5f05b-9357-4a6c-a500-44b45ff5064c"
      },
      "source": [
        "!fastq-dump --split-3 ERR5908782"
      ],
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Rejected 48 READS because of filtering out non-biological READS\n",
            "Read 153205 spots for ERR5908782\n",
            "Written 153205 spots for ERR5908782\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8opOwIMEfYfb"
      },
      "source": [
        "#### Utilizando as ferramentas BWA e Samtools, realize o mapeamento das leituras."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f__unGZ1PiCg",
        "outputId": "97368b7b-3be6-4fea-8ecc-31025f1ee718"
      },
      "source": [
        "!bwa index -p sarscov2 NC_045512.2.fasta"
      ],
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[bwa_index] Pack FASTA... 0.00 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[bwa_index] 0.01 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.00 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.00 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 0.00 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index -p sarscov2 NC_045512.2.fasta\n",
            "[main] Real time: 0.049 sec; CPU: 0.021 sec\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hqkZAQ6jPwJx",
        "outputId": "6eecf41f-d7da-486f-e372-47451912d904"
      },
      "source": [
        "!bwa aln sarscov2 ERR5908782_1.fastq > ERR5908782_1.sai\n",
        "!bwa aln sarscov2 ERR5908782_2.fastq > ERR5908782_2.sai"
      ],
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[bwa_aln] 17bp reads: max_diff = 2\n",
            "[bwa_aln] 38bp reads: max_diff = 3\n",
            "[bwa_aln] 64bp reads: max_diff = 4\n",
            "[bwa_aln] 93bp reads: max_diff = 5\n",
            "[bwa_aln] 124bp reads: max_diff = 6\n",
            "[bwa_aln] 157bp reads: max_diff = 7\n",
            "[bwa_aln] 190bp reads: max_diff = 8\n",
            "[bwa_aln] 225bp reads: max_diff = 9\n",
            "[bwa_aln_core] calculate SA coordinate... 4.50 sec\n",
            "[bwa_aln_core] write to the disk... 0.01 sec\n",
            "[bwa_aln_core] 153157 sequences have been processed.\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa aln sarscov2 ERR5908782_1.fastq\n",
            "[main] Real time: 4.820 sec; CPU: 4.785 sec\n",
            "[bwa_aln] 17bp reads: max_diff = 2\n",
            "[bwa_aln] 38bp reads: max_diff = 3\n",
            "[bwa_aln] 64bp reads: max_diff = 4\n",
            "[bwa_aln] 93bp reads: max_diff = 5\n",
            "[bwa_aln] 124bp reads: max_diff = 6\n",
            "[bwa_aln] 157bp reads: max_diff = 7\n",
            "[bwa_aln] 190bp reads: max_diff = 8\n",
            "[bwa_aln] 225bp reads: max_diff = 9\n",
            "[bwa_aln_core] calculate SA coordinate... 4.47 sec\n",
            "[bwa_aln_core] write to the disk... 0.02 sec\n",
            "[bwa_aln_core] 153157 sequences have been processed.\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa aln sarscov2 ERR5908782_2.fastq\n",
            "[main] Real time: 4.755 sec; CPU: 4.740 sec\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "r65jV7Y2QnPD",
        "outputId": "c9937b10-e5c0-42d2-d56e-fc99d7787a4d"
      },
      "source": [
        "!bwa sampe sarscov2 ERR5908782_1.sai ERR5908782_2.sai ERR5908782_1.fastq ERR5908782_2.fastq > alignment.sam"
      ],
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[bwa_sai2sam_pe_core] convert to sequence coordinate... \n",
            "[infer_isize] (25, 50, 75) percentile: (94, 138, 198)\n",
            "[infer_isize] low and high boundaries: 200 and 406 for estimating avg and std\n",
            "[infer_isize] inferred external isize from 36138 pairs: 257.792 +/- 44.163\n",
            "[infer_isize] skewness: 0.730; kurtosis: -0.318; ap_prior: 1.00e-05\n",
            "[infer_isize] inferred maximum insert size: 483 (5.11 sigma)\n",
            "[bwa_sai2sam_pe_core] time elapses: 0.87 sec\n",
            "[bwa_sai2sam_pe_core] changing coordinates of 44 alignments.\n",
            "[bwa_sai2sam_pe_core] align unmapped mate...\n",
            "[bwa_paired_sw] 751 out of 2000 Q17 singletons are mated.\n",
            "[bwa_paired_sw] 2 out of 112165 Q17 discordant pairs are fixed.\n",
            "[bwa_sai2sam_pe_core] time elapses: 2.95 sec\n",
            "[bwa_sai2sam_pe_core] refine gapped alignments... 0.57 sec\n",
            "[bwa_sai2sam_pe_core] print alignments... 0.66 sec\n",
            "[bwa_sai2sam_pe_core] 153157 sequences have been processed.\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa sampe sarscov2 ERR5908782_1.sai ERR5908782_2.sai ERR5908782_1.fastq ERR5908782_2.fastq\n",
            "[main] Real time: 7.251 sec; CPU: 5.702 sec\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "o6KYio1jRdNk",
        "outputId": "0d1efd43-45e3-4447-8af1-1610f93ee2b4"
      },
      "source": [
        "!head -10 alignment.sam"
      ],
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "@SQ\tSN:NC_045512.2\tLN:29903\n",
            "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa sampe sarscov2 ERR5908782_1.sai ERR5908782_2.sai ERR5908782_1.fastq ERR5908782_2.fastq\n",
            "ERR5908782.1\t81\tNC_045512.2\t13654\t37\t83M\t=\t13654\t-83\tGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAAGGATTGTCCAGCTGTTGCTAAACA\tGGGFGGGGGGGFGGGFCGDGGGGGGGFGGGGGGGDGGFFEGGGGFGGGGGGGGGGGFGF@GGGGGGGGGGGGGGGGGGCCCCC\tXT:A:U\tNM:i:0\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tMD:Z:83\n",
            "ERR5908782.1\t161\tNC_045512.2\t13654\t37\t83M\t=\t13654\t83\tGTTAAGAGACACACTTTCTCTAACTACCAACATGAAGAAACAATTTATAATTTACTTAAGGATTGTCCAGCTGTTGCTAAACA\tCCCCCGGFFEEGGGGGCEFFGGFGGA8CEEF,EAFFGFEFGGGGFEGFGGGGGGGGFGGGF<FGDGFGGGGFEGDGGCGGDEF\tXT:A:U\tNM:i:0\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tMD:Z:83\n",
            "ERR5908782.2\t99\tNC_045512.2\t23284\t60\t195M\t=\t23315\t231\tTGCTGTCCGTGATCCACAGACACTTGAGATTCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTAACCAGGTTGCTGTTCTTTATCAGGGTGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGG\tGGGGGAEFGFDCEFFGGGGGGFGGGGGGFFGFFGCAFFGG<FFFDFFFGGFGAFGFGGDFGGGDCDGGGGGGGGGGGGGFG,CFG<CFEFFGGGGFFFGADGGGGGGE<EFGGDGCEFGGGGGGGFFFGDFGAFGFDG<AAFGEEFCGFFGGDGG9FGFFF=FGGGGGGAFFFGFGEEGGGGGGFFCFDFGF>EF\tXT:A:U\tNM:i:1\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tMD:Z:119A75\n",
            "ERR5908782.2\t147\tNC_045512.2\t23315\t60\t200M\t=\t23284\t-231\tCTTGACATTACACCATGTTCTTTTGGTGGTGTCAGTGTTATAACACCAGGAACAAATACTTCTAACCAGGTTGCTGTTCTTTATCAGGGTGTTAACTGCACAGAAGTCCCTGTTGCTATTCATGCAGATCAACTTACTCCTACTTGGCGTGTTTATTCTACAGGTTCTAATGTTTTTCAAACACGTGCAGGCTGTTTAAT\t,DFGFD7,FE9FCCFCFED@,?GGGFDEEE8GGGGGGEFF=GF:DCGGEGFF9CEAAEFEFFFF9A<F<EFCGDGGFE<?E<FB@DCBCGGFF?@FGFCEFGGFEDB7,CCFGGFFCGGDCFCFFCFEFCFGGDFCEGGGFFFF@@ECGFFAFFGGGFFCECEFGDFGGFGGFGGGGGGGGGFCGDGGGFECFDF@CCCC\tXT:A:U\tNM:i:1\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tMD:Z:88A111\n",
            "ERR5908782.3\t97\tNC_045512.2\t24901\t37\t158M\t=\t24901\t158\tAATCATTACTACACACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAATTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGATT\tCCCCCGGGGGGGFGGGEGGGGGGGGGGGGGGGGGGGGGGGGGGFFGGGGFGGFGGGGGGGGGGGGGGGGGGGGGGGGGDGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGFGG:CDGGFGGGFGGGGFFGGFFDGGGGGGGGGGGGGGGFGGGGGGGG\tXT:A:U\tNM:i:1\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tMD:Z:13G144\n",
            "ERR5908782.3\t145\tNC_045512.2\t24901\t37\t158M\t=\t24901\t-158\tAATCATTACTACACACAACACATTTGTGTCTGGTAACTGTGATGTTGTAATAGGAATTGTCAACAACACAGTTTATGATCCTTTGCAACCTGAATTAGACTCATTCAAGGAGGAGTTAGATAAATATTTTAAGAATCATACATCACCAGATGTTGATT\tF9AGGGGGGGGGGGGGGFCFGFGGGGGGGGGFGGGGFFGGGGGGGFGGGFFGGGGGGGGGGGGFGGGGGGGFFGGGGDGGGGGGGGGGGGGGGGCGGGGGGGGGGGGGGGGGGGGFFGFGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGCCCCC\tXT:A:U\tNM:i:1\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tMD:Z:13G144\n",
            "ERR5908782.4\t97\tNC_045512.2\t23903\t37\t174M\t=\t23903\t174\tGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCTACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGT\tCCCCCGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGGFGGFFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGFGG\tXT:A:U\tNM:i:0\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tMD:Z:174\n",
            "ERR5908782.4\t145\tNC_045512.2\t23903\t37\t174M\t=\t23903\t-174\tGTTTTTGCACAAGTCAAACAAATTTACAAAACACCACCAATTAAAGATTTTGGTGGTTTTAATTTTTCACAAATATTACCAGATCCATCAAAACCAAGCAAGAGGTCATTTATTGAAGATCTACTTTTCAACAAAGTGACACTTGCAGATGCTGGCTTCATCAAACAATATGGT\tGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGDGGGGGGFAGGGGGGGCFGGGGGGF<GGGGGCGGGGGGGGGGGGGGGGGFGGGGEDGGGGGGGGGGGGGGGGGGGGGGGFFGGGGGGGGGGFGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGCCCCC\tXT:A:U\tNM:i:0\tSM:i:37\tAM:i:37\tX0:i:1\tX1:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tMD:Z:174\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "XyEPFLvPRqnZ"
      },
      "source": [
        "!samtools view -S -b -o alignment.bam alignment.sam"
      ],
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OGGwzEd9R8zS",
        "outputId": "cef8eb95-8ac7-4f61-b4b9-4fb63e1c4556"
      },
      "source": [
        "!samtools view -h"
      ],
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\n",
            "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
            "\n",
            "Options:\n",
            "  -b       output BAM\n",
            "  -C       output CRAM (requires -T)\n",
            "  -1       use fast BAM compression (implies -b)\n",
            "  -u       uncompressed BAM output (implies -b)\n",
            "  -h       include header in SAM output\n",
            "  -H       print SAM header only (no alignments)\n",
            "  -c       print only the count of matching records\n",
            "  -o FILE  output file name [stdout]\n",
            "  -U FILE  output reads not selected by filters to FILE [null]\n",
            "  -t FILE  FILE listing reference names and lengths (see long help) [null]\n",
            "  -L FILE  only include reads overlapping this BED FILE [null]\n",
            "  -r STR   only include reads in read group STR [null]\n",
            "  -R FILE  only include reads with read group listed in FILE [null]\n",
            "  -q INT   only include reads with mapping quality >= INT [0]\n",
            "  -l STR   only include reads in library STR [null]\n",
            "  -m INT   only include reads with number of CIGAR operations consuming\n",
            "           query sequence >= INT [0]\n",
            "  -f INT   only include reads with all  of the FLAGs in INT present [0]\n",
            "  -F INT   only include reads with none of the FLAGS in INT present [0]\n",
            "  -G INT   only EXCLUDE reads with all  of the FLAGs in INT present [0]\n",
            "  -s FLOAT subsample reads (given INT.FRAC option value, 0.FRAC is the\n",
            "           fraction of templates/read pairs to keep; INT part sets seed)\n",
            "  -M       use the multi-region iterator (increases the speed, removes\n",
            "           duplicates and outputs the reads as they are ordered in the file)\n",
            "  -x STR   read tag to strip (repeatable) [null]\n",
            "  -B       collapse the backward CIGAR operation\n",
            "  -?       print long help, including note about region specification\n",
            "  -S       ignored (input format is auto-detected)\n",
            "      --input-fmt-option OPT[=VAL]\n",
            "               Specify a single input file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
            "               Specify output format (SAM, BAM, CRAM)\n",
            "      --output-fmt-option OPT[=VAL]\n",
            "               Specify a single output file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -T, --reference FILE\n",
            "               Reference sequence FASTA FILE [null]\n",
            "  -@, --threads INT\n",
            "               Number of additional threads to use [0]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "7JCLxnMHSqY6"
      },
      "source": [
        "!samtools sort -o alignment.sorted.bam alignment.bam"
      ],
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "p5QnccafflJx"
      },
      "source": [
        "#### Utilize as ferramentas Bcftools e Vcfutils para identificar as variantes genéticas a partir do arquivo BAM (ordenado)."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YkJq_K3oTvl_",
        "outputId": "097f4a5a-b301-4691-ef85-f8babcc26a4f"
      },
      "source": [
        "!bcftools mpileup -f NC_045512.2.fasta -O z -o alignment.vcf.gz -m 10 alignment.sorted.bam"
      ],
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[mpileup] 1 samples in 1 input files\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "37giUZDaVGaa"
      },
      "source": [
        "!bcftools call -c -v --ploidy 1 alignment.vcf.gz > variants.raw.vcf"
      ],
      "execution_count": 14,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tTOHlR8TX1JM",
        "outputId": "da0b1d38-83de-4ba4-a02a-77fe0ea7fce5"
      },
      "source": [
        "!head -50 variants.raw.vcf"
      ],
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "##fileformat=VCFv4.2\n",
            "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
            "##bcftoolsVersion=1.7+htslib-1.7-2\n",
            "##bcftoolsCommand=mpileup -f NC_045512.2.fasta -O z -o alignment.vcf.gz -m 10 alignment.sorted.bam\n",
            "##reference=file://NC_045512.2.fasta\n",
            "##contig=<ID=NC_045512.2,length=29903>\n",
            "##ALT=<ID=*,Description=\"Represents allele(s) other than observed.\">\n",
            "##INFO=<ID=INDEL,Number=0,Type=Flag,Description=\"Indicates that the variant is an INDEL.\">\n",
            "##INFO=<ID=IDV,Number=1,Type=Integer,Description=\"Maximum number of reads supporting an indel\">\n",
            "##INFO=<ID=IMF,Number=1,Type=Float,Description=\"Maximum fraction of reads supporting an indel\">\n",
            "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Raw read depth\">\n",
            "##INFO=<ID=VDB,Number=1,Type=Float,Description=\"Variant Distance Bias for filtering splice-site artefacts in RNA-seq data (bigger is better)\",Version=\"3\">\n",
            "##INFO=<ID=RPB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Read Position Bias (bigger is better)\">\n",
            "##INFO=<ID=MQB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Mapping Quality Bias (bigger is better)\">\n",
            "##INFO=<ID=BQB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Base Quality Bias (bigger is better)\">\n",
            "##INFO=<ID=MQSB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Mapping Quality vs Strand Bias (bigger is better)\">\n",
            "##INFO=<ID=SGB,Number=1,Type=Float,Description=\"Segregation based metric.\">\n",
            "##INFO=<ID=MQ0F,Number=1,Type=Float,Description=\"Fraction of MQ0 reads (smaller is better)\">\n",
            "##FORMAT=<ID=PL,Number=G,Type=Integer,Description=\"List of Phred-scaled genotype likelihoods\">\n",
            "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
            "##INFO=<ID=AF1,Number=1,Type=Float,Description=\"Max-likelihood estimate of the first ALT allele frequency (assuming HWE)\">\n",
            "##INFO=<ID=AF2,Number=1,Type=Float,Description=\"Max-likelihood estimate of the first and second group ALT allele frequency (assuming HWE)\">\n",
            "##INFO=<ID=AC1,Number=1,Type=Float,Description=\"Max-likelihood estimate of the first ALT allele count (no HWE assumption)\">\n",
            "##INFO=<ID=MQ,Number=1,Type=Integer,Description=\"Root-mean-square mapping quality of covering reads\">\n",
            "##INFO=<ID=FQ,Number=1,Type=Float,Description=\"Phred probability of all samples being the same\">\n",
            "##INFO=<ID=PV4,Number=4,Type=Float,Description=\"P-values for strand bias, baseQ bias, mapQ bias and tail distance bias\">\n",
            "##INFO=<ID=G3,Number=3,Type=Float,Description=\"ML estimate of genotype frequencies\">\n",
            "##INFO=<ID=HWE,Number=1,Type=Float,Description=\"Chi^2 based HWE test P-value based on G3\">\n",
            "##INFO=<ID=DP4,Number=4,Type=Integer,Description=\"Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases\">\n",
            "##bcftools_callVersion=1.7+htslib-1.7-2\n",
            "##bcftools_callCommand=call -c -v --ploidy 1 alignment.vcf.gz; Date=Mon May 24 12:35:18 2021\n",
            "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\talignment.sorted.bam\n",
            "NC_045512.2\t241\t.\tC\tT\t225.007\t.\tDP=203;VDB=7.87795e-06;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,101,2;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t913\t.\tC\tT\t225.007\t.\tDP=91;VDB=8.58553e-05;SGB=-0.693147;MQSB=0.978481;MQ0F=0;AF1=1;AC1=1;DP4=0,0,44,37;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t2698\t.\tT\tC\t225.007\t.\tDP=299;VDB=2.07017e-05;SGB=-0.693147;RPB=0.287879;MQB=0.99495;MQSB=0.999848;BQB=0.785354;MQ0F=0;AF1=1;AC1=1;DP4=2,0,194,4;MQ=60;FQ=-999;PV4=1,1,0.460122,0.173563\tGT:PL\t1:255,0\n",
            "NC_045512.2\t3037\t.\tC\tT\t225.007\t.\tDP=302;VDB=0.000148787;SGB=-0.693147;MQSB=0.975901;MQ0F=0;AF1=1;AC1=1;DP4=0,0,196,23;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t3267\t.\tC\tT\t225.007\t.\tDP=300;VDB=1.09469e-37;SGB=-0.693147;MQSB=0.99944;MQ0F=0;AF1=1;AC1=1;DP4=0,0,250,6;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t4668\t.\tC\tT\t225.007\t.\tDP=152;VDB=0;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,135,2;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t4964\t.\tA\tG\t225.007\t.\tDP=40;VDB=2.12242e-07;SGB=-0.69312;RPB=1;MQB=1;MQSB=1;BQB=1;MQ0F=0;AF1=1;AC1=1;DP4=1,0,15,17;MQ=60;FQ=-999;PV4=0.484848,1,1,1\tGT:PL\t1:255,0\n",
            "NC_045512.2\t5388\t.\tC\tA\t225.007\t.\tDP=290;VDB=1.89403e-39;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,228,6;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t5986\t.\tC\tT\t225.007\t.\tDP=271;VDB=3.26051e-40;SGB=-0.693147;MQSB=0.867015;MQ0F=0;AF1=1;AC1=1;DP4=0,0,215,10;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t6954\t.\tT\tC\t225.007\t.\tDP=111;VDB=0.00356479;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,61,36;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t11287\t.\tGTCTGGTTTT\tG\t217.469\t.\tINDEL;IDV=154;IMF=0.641667;DP=240;VDB=0.384104;SGB=-0.693147;MQSB=0.527818;MQ0F=0;AF1=1;AC1=1;DP4=2,78,7,153;MQ=29;FQ=-999;PV4=0.721614,1,0.158655,1\tGT:PL\t1:255,38,0\n",
            "NC_045512.2\t12162\t.\tA\tG\t225.007\t.\tDP=250;VDB=0.114677;SGB=-0.693147;RPB=0.760274;MQB=0.990868;MQSB=0.996403;BQB=0.0776256;MQ0F=0;AF1=1;AC1=1;DP4=2,0,193,26;MQ=60;FQ=-999;PV4=1,1,0.446307,1\tGT:PL\t1:255,0\n",
            "NC_045512.2\t14408\t.\tC\tT\t225.007\t.\tDP=308;VDB=6.79806e-09;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,230,15;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t14676\t.\tC\tT\t225.007\t.\tDP=299;VDB=0;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,267,6;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t15240\t.\tC\tT\t225.007\t.\tDP=255;VDB=0;SGB=-0.693147;RPB=1;MQB=1;MQSB=1;BQB=1;MQ0F=0;AF1=1;AC1=1;DP4=1,0,220,3;MQ=60;FQ=-999;PV4=1,1,1,0.41414\tGT:PL\t1:255,0\n",
            "NC_045512.2\t15279\t.\tC\tT\t225.007\t.\tDP=279;VDB=0;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,236,3;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t16176\t.\tT\tC\t225.007\t.\tDP=122;VDB=0.010996;SGB=-0.693147;MQSB=0.860708;MQ0F=0;AF1=1;AC1=1;DP4=0,0,90,9;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t17615\t.\tA\tG\t225.007\t.\tDP=208;VDB=0.000591934;SGB=-0.693147;RPB=0.87987;MQB=1;MQSB=1;BQB=0.548701;MQ0F=0;AF1=1;AC1=1;DP4=2,0,123,31;MQ=60;FQ=-999;PV4=1,0.0698796,1,0.455969\tGT:PL\t1:255,0\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ASucqdaUYY_C"
      },
      "source": [
        "!vcfutils.pl varFilter -d 30 variants.raw.vcf > variants.filtered.vcf"
      ],
      "execution_count": 16,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iJAUaqcnZdEb",
        "outputId": "2db474ca-a999-43c6-8d63-13e4c90acfe9"
      },
      "source": [
        "!head -50 variants.filtered.vcf"
      ],
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "##fileformat=VCFv4.2\n",
            "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
            "##bcftoolsVersion=1.7+htslib-1.7-2\n",
            "##bcftoolsCommand=mpileup -f NC_045512.2.fasta -O z -o alignment.vcf.gz -m 10 alignment.sorted.bam\n",
            "##reference=file://NC_045512.2.fasta\n",
            "##contig=<ID=NC_045512.2,length=29903>\n",
            "##ALT=<ID=*,Description=\"Represents allele(s) other than observed.\">\n",
            "##INFO=<ID=INDEL,Number=0,Type=Flag,Description=\"Indicates that the variant is an INDEL.\">\n",
            "##INFO=<ID=IDV,Number=1,Type=Integer,Description=\"Maximum number of reads supporting an indel\">\n",
            "##INFO=<ID=IMF,Number=1,Type=Float,Description=\"Maximum fraction of reads supporting an indel\">\n",
            "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Raw read depth\">\n",
            "##INFO=<ID=VDB,Number=1,Type=Float,Description=\"Variant Distance Bias for filtering splice-site artefacts in RNA-seq data (bigger is better)\",Version=\"3\">\n",
            "##INFO=<ID=RPB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Read Position Bias (bigger is better)\">\n",
            "##INFO=<ID=MQB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Mapping Quality Bias (bigger is better)\">\n",
            "##INFO=<ID=BQB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Base Quality Bias (bigger is better)\">\n",
            "##INFO=<ID=MQSB,Number=1,Type=Float,Description=\"Mann-Whitney U test of Mapping Quality vs Strand Bias (bigger is better)\">\n",
            "##INFO=<ID=SGB,Number=1,Type=Float,Description=\"Segregation based metric.\">\n",
            "##INFO=<ID=MQ0F,Number=1,Type=Float,Description=\"Fraction of MQ0 reads (smaller is better)\">\n",
            "##FORMAT=<ID=PL,Number=G,Type=Integer,Description=\"List of Phred-scaled genotype likelihoods\">\n",
            "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
            "##INFO=<ID=AF1,Number=1,Type=Float,Description=\"Max-likelihood estimate of the first ALT allele frequency (assuming HWE)\">\n",
            "##INFO=<ID=AF2,Number=1,Type=Float,Description=\"Max-likelihood estimate of the first and second group ALT allele frequency (assuming HWE)\">\n",
            "##INFO=<ID=AC1,Number=1,Type=Float,Description=\"Max-likelihood estimate of the first ALT allele count (no HWE assumption)\">\n",
            "##INFO=<ID=MQ,Number=1,Type=Integer,Description=\"Root-mean-square mapping quality of covering reads\">\n",
            "##INFO=<ID=FQ,Number=1,Type=Float,Description=\"Phred probability of all samples being the same\">\n",
            "##INFO=<ID=PV4,Number=4,Type=Float,Description=\"P-values for strand bias, baseQ bias, mapQ bias and tail distance bias\">\n",
            "##INFO=<ID=G3,Number=3,Type=Float,Description=\"ML estimate of genotype frequencies\">\n",
            "##INFO=<ID=HWE,Number=1,Type=Float,Description=\"Chi^2 based HWE test P-value based on G3\">\n",
            "##INFO=<ID=DP4,Number=4,Type=Integer,Description=\"Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases\">\n",
            "##bcftools_callVersion=1.7+htslib-1.7-2\n",
            "##bcftools_callCommand=call -c -v --ploidy 1 alignment.vcf.gz; Date=Mon May 24 12:35:18 2021\n",
            "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\talignment.sorted.bam\n",
            "NC_045512.2\t241\t.\tC\tT\t225.007\t.\tDP=203;VDB=7.87795e-06;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,101,2;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t913\t.\tC\tT\t225.007\t.\tDP=91;VDB=8.58553e-05;SGB=-0.693147;MQSB=0.978481;MQ0F=0;AF1=1;AC1=1;DP4=0,0,44,37;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t2698\t.\tT\tC\t225.007\t.\tDP=299;VDB=2.07017e-05;SGB=-0.693147;RPB=0.287879;MQB=0.99495;MQSB=0.999848;BQB=0.785354;MQ0F=0;AF1=1;AC1=1;DP4=2,0,194,4;MQ=60;FQ=-999;PV4=1,1,0.460122,0.173563\tGT:PL\t1:255,0\n",
            "NC_045512.2\t3037\t.\tC\tT\t225.007\t.\tDP=302;VDB=0.000148787;SGB=-0.693147;MQSB=0.975901;MQ0F=0;AF1=1;AC1=1;DP4=0,0,196,23;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t3267\t.\tC\tT\t225.007\t.\tDP=300;VDB=1.09469e-37;SGB=-0.693147;MQSB=0.99944;MQ0F=0;AF1=1;AC1=1;DP4=0,0,250,6;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t4668\t.\tC\tT\t225.007\t.\tDP=152;VDB=0;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,135,2;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t4964\t.\tA\tG\t225.007\t.\tDP=40;VDB=2.12242e-07;SGB=-0.69312;RPB=1;MQB=1;MQSB=1;BQB=1;MQ0F=0;AF1=1;AC1=1;DP4=1,0,15,17;MQ=60;FQ=-999;PV4=0.484848,1,1,1\tGT:PL\t1:255,0\n",
            "NC_045512.2\t5388\t.\tC\tA\t225.007\t.\tDP=290;VDB=1.89403e-39;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,228,6;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t5986\t.\tC\tT\t225.007\t.\tDP=271;VDB=3.26051e-40;SGB=-0.693147;MQSB=0.867015;MQ0F=0;AF1=1;AC1=1;DP4=0,0,215,10;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t6954\t.\tT\tC\t225.007\t.\tDP=111;VDB=0.00356479;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,61,36;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t11287\t.\tGTCTGGTTTT\tG\t217.469\t.\tINDEL;IDV=154;IMF=0.641667;DP=240;VDB=0.384104;SGB=-0.693147;MQSB=0.527818;MQ0F=0;AF1=1;AC1=1;DP4=2,78,7,153;MQ=29;FQ=-999;PV4=0.721614,1,0.158655,1\tGT:PL\t1:255,38,0\n",
            "NC_045512.2\t12162\t.\tA\tG\t225.007\t.\tDP=250;VDB=0.114677;SGB=-0.693147;RPB=0.760274;MQB=0.990868;MQSB=0.996403;BQB=0.0776256;MQ0F=0;AF1=1;AC1=1;DP4=2,0,193,26;MQ=60;FQ=-999;PV4=1,1,0.446307,1\tGT:PL\t1:255,0\n",
            "NC_045512.2\t14408\t.\tC\tT\t225.007\t.\tDP=308;VDB=6.79806e-09;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,230,15;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t14676\t.\tC\tT\t225.007\t.\tDP=299;VDB=0;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,267,6;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t15240\t.\tC\tT\t225.007\t.\tDP=255;VDB=0;SGB=-0.693147;RPB=1;MQB=1;MQSB=1;BQB=1;MQ0F=0;AF1=1;AC1=1;DP4=1,0,220,3;MQ=60;FQ=-999;PV4=1,1,1,0.41414\tGT:PL\t1:255,0\n",
            "NC_045512.2\t15279\t.\tC\tT\t225.007\t.\tDP=279;VDB=0;SGB=-0.693147;MQSB=1;MQ0F=0;AF1=1;AC1=1;DP4=0,0,236,3;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t16176\t.\tT\tC\t225.007\t.\tDP=122;VDB=0.010996;SGB=-0.693147;MQSB=0.860708;MQ0F=0;AF1=1;AC1=1;DP4=0,0,90,9;MQ=60;FQ=-999\tGT:PL\t1:255,0\n",
            "NC_045512.2\t17615\t.\tA\tG\t225.007\t.\tDP=208;VDB=0.000591934;SGB=-0.693147;RPB=0.87987;MQB=1;MQSB=1;BQB=0.548701;MQ0F=0;AF1=1;AC1=1;DP4=2,0,123,31;MQ=60;FQ=-999;PV4=1,0.0698796,1,0.455969\tGT:PL\t1:255,0\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pCfMbiB_fs2T"
      },
      "source": [
        "#### Utilize a ferramenta Snptoolkit para anotar as variantes com base na anotação do genoma."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eKFL_OyJZv6p",
        "outputId": "a2bd8f0c-3db4-47f4-e432-3ce831610a00"
      },
      "source": [
        "!snptoolkit annotate -i variants.filtered.vcf -g NC_045512.2.gb"
      ],
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32m[12:55:42]\u001b[0m \u001b[37;40m[INFO]\u001b[0m [1 CPUs requested out of 2 detected on this machine]\n",
            "\u001b[32m[12:55:42]\u001b[0m \u001b[37;40m[INFO]\u001b[0m [snpToolkit is filtering and annotating your SNPs]\n",
            "\r  0% 0/1 [00:00<?, ?it/s]\u001b[32m[12:55:42]\u001b[0m \u001b[37;40m[INFO]\u001b[0m [snpToolkit output files for SNPs data will be located in folders snpToolkit_SNPs_output_2021-05-24_at_12.55.42]\n",
            "\u001b[32m[12:55:42]\u001b[0m \u001b[37;40m[INFO]\u001b[0m [snpToolkit output files for INDELs data will be located in folders snpToolkit_INDELS_output_2021-05-24_at_12.55.42]\n",
            "\r100% 1/1 [00:00<00:00,  7.41it/s]\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1c8T3qStfxY5"
      },
      "source": [
        "#### Analisando os dados de SNPs"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 501
        },
        "id": "dKSxqJ8Zcl9O",
        "outputId": "1d01a93b-0f3b-4d1f-d22d-65f0edb8587b"
      },
      "source": [
        "import pandas as pd\n",
        "snps_df = pd.read_excel('snps_sarscov2.xlsx')\n",
        "snps_df.head()"
      ],
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>##Coordinates</th>\n",
              "      <th>REF</th>\n",
              "      <th>SNP</th>\n",
              "      <th>Depth</th>\n",
              "      <th>Nb of reads REF</th>\n",
              "      <th>Nb reads SNPs</th>\n",
              "      <th>Ratio</th>\n",
              "      <th>Quality</th>\n",
              "      <th>Annotation</th>\n",
              "      <th>Product</th>\n",
              "      <th>Orientation</th>\n",
              "      <th>Coordinates in gene</th>\n",
              "      <th>Ref codon</th>\n",
              "      <th>SNP codon</th>\n",
              "      <th>Ref AA</th>\n",
              "      <th>SNP AA</th>\n",
              "      <th>Coordinates protein</th>\n",
              "      <th>Effect</th>\n",
              "      <th>Location</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>241</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>203</td>\n",
              "      <td>0</td>\n",
              "      <td>103</td>\n",
              "      <td>1.00</td>\n",
              "      <td>225007</td>\n",
              "      <td>intergenic</td>\n",
              "      <td>.</td>\n",
              "      <td>+</td>\n",
              "      <td>25 bp from GU280_gp01|ORF1ab</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>NC_045512.2: Severe acute respiratory syndrome...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>913</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>91</td>\n",
              "      <td>0</td>\n",
              "      <td>81</td>\n",
              "      <td>1.00</td>\n",
              "      <td>225007</td>\n",
              "      <td>GU280_gp01|ORF1ab</td>\n",
              "      <td>ORF1a polyprotein</td>\n",
              "      <td>+</td>\n",
              "      <td>648</td>\n",
              "      <td>TCC</td>\n",
              "      <td>TC[T]</td>\n",
              "      <td>S</td>\n",
              "      <td>S</td>\n",
              "      <td>216</td>\n",
              "      <td>Syn</td>\n",
              "      <td>NC_045512.2: Severe acute respiratory syndrome...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2698</td>\n",
              "      <td>T</td>\n",
              "      <td>C</td>\n",
              "      <td>299</td>\n",
              "      <td>2</td>\n",
              "      <td>198</td>\n",
              "      <td>0.99</td>\n",
              "      <td>225007</td>\n",
              "      <td>GU280_gp01|ORF1ab</td>\n",
              "      <td>ORF1a polyprotein</td>\n",
              "      <td>+</td>\n",
              "      <td>2433</td>\n",
              "      <td>AAT</td>\n",
              "      <td>AA[C]</td>\n",
              "      <td>N</td>\n",
              "      <td>N</td>\n",
              "      <td>811</td>\n",
              "      <td>Syn</td>\n",
              "      <td>NC_045512.2: Severe acute respiratory syndrome...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>3037</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>302</td>\n",
              "      <td>0</td>\n",
              "      <td>219</td>\n",
              "      <td>1.00</td>\n",
              "      <td>225007</td>\n",
              "      <td>GU280_gp01|ORF1ab</td>\n",
              "      <td>ORF1a polyprotein</td>\n",
              "      <td>+</td>\n",
              "      <td>2772</td>\n",
              "      <td>TTC</td>\n",
              "      <td>TT[T]</td>\n",
              "      <td>F</td>\n",
              "      <td>F</td>\n",
              "      <td>924</td>\n",
              "      <td>Syn</td>\n",
              "      <td>NC_045512.2: Severe acute respiratory syndrome...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>3267</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>300</td>\n",
              "      <td>0</td>\n",
              "      <td>256</td>\n",
              "      <td>1.00</td>\n",
              "      <td>225007</td>\n",
              "      <td>GU280_gp01|ORF1ab</td>\n",
              "      <td>ORF1a polyprotein</td>\n",
              "      <td>+</td>\n",
              "      <td>3002</td>\n",
              "      <td>ACT</td>\n",
              "      <td>A[T]T</td>\n",
              "      <td>T</td>\n",
              "      <td>I</td>\n",
              "      <td>1001</td>\n",
              "      <td>NS</td>\n",
              "      <td>NC_045512.2: Severe acute respiratory syndrome...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   ##Coordinates REF  ... Effect                                           Location\n",
              "0            241   C  ...      .  NC_045512.2: Severe acute respiratory syndrome...\n",
              "1            913   C  ...    Syn  NC_045512.2: Severe acute respiratory syndrome...\n",
              "2           2698   T  ...    Syn  NC_045512.2: Severe acute respiratory syndrome...\n",
              "3           3037   C  ...    Syn  NC_045512.2: Severe acute respiratory syndrome...\n",
              "4           3267   C  ...     NS  NC_045512.2: Severe acute respiratory syndrome...\n",
              "\n",
              "[5 rows x 19 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 19
        }
      ]
    }
  ]
}